### PR TITLE
Fixed #8936 -- Added view permissions and a read-only admin

### PR DESCRIPTION
**Pull request by PetrDlouhy**

### CODE REVIEW

File: django/contrib/admin/sites.py [hunk 1]
```diff
@@ -443,7 +443,7 @@ def _build_app_dict(self, request, label=None):
                 'object_name': model._meta.object_name,
                 'perms': perms,
             }
-            if perms.get('change'):
+            if perms.get('change', False) or perms.get('view', False):
```
Comment (charettes) [inline django/contrib/admin/sites.py]: No need to use `False` as a default value here, `None` also evaluates to `False`.

File: django/contrib/admin/options.py [hunk 2]
```diff
@@ -307,6 +307,22 @@ def get_readonly_fields(self, request, obj=None):
         """
         return self.readonly_fields
 
+    def get_uneditable_fields(self, request, obj=None):
+        """
+        Get fields that are either readonly or hasn't change/add permissions
+        """
+        if hasattr(request, 'user') and (
```
Comment (charettes) [inline django/contrib/admin/options.py]: How can you reach this point with no `request.user`?

File: django/contrib/admin/options.py [hunk 3]
```diff
@@ -307,6 +307,22 @@ def get_readonly_fields(self, request, obj=None):
...
+                self.has_change_permission(request, obj) or (self.has_add_permission(request) and not obj)):
+            return self.get_readonly_fields(request, obj)
+        else:
+            if hasattr(self, 'declared_fieldsets') and self.declared_fieldsets:
```
Comment (charettes) [inline django/contrib/admin/options.py]: Use `elif` here?

File: django/contrib/admin/options.py (hunk 2 above)
Comment (PetrDlouhy) [inline django/contrib/admin/options.py]: It happen. The tests fail without this test.

**Comment by PetrDlouhy** [comment]
@charettes: Thanks for the suggestions.
I accidentally deleted the original PR branch and then messed the PRs a little bit. The active PR now is #5297, where are I made the corrections.

**Pull request closed without merging**